projects/storefrontlib/shared/components/hierarchy/hierarchy-node-title/hierarchy-node-title.component.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, Input } from '@angular/core';
+import { TitleNode } from './title-node.model';
+import { ActiveCartFacade } from '@spartacus/cart/base/root';
+
+/**
+ * Hierarchy Selection node variant to show a title
+ */
+@Component({
+  selector: 'cx-hierarchy-node-title',
+  templateUrl: './hierarchy-node-title.component.html',
+})
+export class HierarchyNodeTitleComponent {
+  @Input() tree: TitleNode;
+
+  @Input() activeCartService: ActiveCartFacade;
+
+  @Input() readonly = false;
+
+  removeBundle(entryGroupNumber: any) {
```

Comment:
I see here that this is a very targeted purpose for the `onClick`. I have ellaborated further in another comment of whether this component is "cart-specific" or "dumb". But if you were to go down the dumb component route, you can pass some `doOnClick()` method as an input so that you can declare it earlier.

For example:

```
@Input() doOnClick: Function
...
onClick() {
  doOnClick()
}
...
<button (click)="onClick()">
...
```

Reply:
Got it, I have made the processing logic of `remove` and `edit` into parameters to receive them with `HierarchyOptions`. If there are new methods in the future, they can also be extended in `HierarchyOptions`. like this：
```
export interface HierarchyOptions{
...
    onItemRemove?: (itemNumber: any) => void; // Function to handle item removal
    onItemEdit?: (itemNumber: any) => void; // Function to handle item edit
...
}
```